File: CSharp-SQLite-APP/SQLiteFunctions/SQLiteFunctions/SQLiteFunctions/Functions/Bin2Dec.cs
using System;
using System.Data.SQLite;

namespace SQLiteFunctions.Functions
{
    [SQLiteFunction(Name = "bin2dec", Arguments = 1, FuncType = FunctionType.Scalar)]
    public class Bin2Dec : SQLiteFunction
    {
        public string ConstructCommand()
        {
            Console.Write("Binary value: ");
            var input = Console.ReadLine();

            return $"bin2dec({input})";
        }

        public override object Invoke(object[] args)
        {
            return BinToDecimal(args[0].ToString());
        }

        private int BinToDecimal(string bin)
        {
            int value = 0;
            int counter = 1;
            for (int i = bin.Length - 1; i >= 0; i--)
            {
                if (i == bin.Length - 1)
                {
                    if (bin[i] == '1') value = 1;
                    counter *= 2;
                    continue;
                }
                if (bin[i] == '1') value += counter;
                counter *= 2;
            }

            return value;
        }
    }
}

File: Cpp-FirebirdSQL/bin2dec/bin2dec.cpp
#include <string>
#include <iostream>

using namespace std;

int bin_to_decimal(string bin)
{
    int value = 0;
    int counter = 1;

    for(int i = bin.length() - 1; i >= 0; i--)
    {
        if(i == bin.length() - 1)
        {
            if(bin[i] == '1') value = 1;
            counter *= 2;
            continue;
        }
        if(bin[i] == '1') value += counter;
        counter *= 2;
    }

    return value;
}

int main()
{
    cout << bin_to_decimal("0") << endl;
}